feat(my-account): add delete payment method modal confirmation

rbcorrales opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds a confirmation flow before removing saved payment methods. Each delete action now opens a modal with the card brand/last four/expiry, and now all the dropdowns close once the modal appears (including address-related modals).

<img width="430" height="381" alt="image" src="https://github.com/user-attachments/assets/49e73977-83cc-4db0-b7b8-259b4065e2f4" />


Closes NPPD-909.

### How to test the changes in this Pull Request:

1. Log in as a reader with multiple saved payment methods and visit My Account → Payment information.
2. Open the overflow menu for any deletable card and click "Delete payment method"
3. Confirm a modal appears with the corresponding brand/last four/expiry and that the dropdown (three-dot menu) automatically closes.
4. Click "Cancel" and verify that the card remains and that the menu items can be selected again.
5. Reopen the modal for the same card, click "Delete payment method" and make sure the card is removed after the page refresh.
6. Test setting up a different default payment method. The dropdown menu should be kept open, but with the options deactivated while the page refreshes (same behavior as before).

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

rbcorrales committed:
feat(my-account): add delete payment method modal confirmation

rbcorrales committed:
fix(my-account): correct modal ID handling

rbcorrales committed:
fix(my-account): remove active class from dropdowns on modal open

miguelpeixe reviewed (changes requested):
Sorry, accidentally approved.

After confirming deletion, the payment method doesn't get deleted, and I get the following error:

<img width="907" height="390" alt="Image" src="https://github.com/user-attachments/assets/a3849120-453e-441c-a781-39cb20d86030" />

This is not an issue in trunk (without the modal).

rbcorrales committed:
fix(my-account): handle query params in form

rbcorrales commented:
@miguelpeixe Thanks for flagging this! Seems that I broke it when converting the URL to a form action, and the nonce was being stripped. This is due to how forms with GET act as per the HTML spec (all the query parameters get removed from the action and replaced with whatever is set via hidden fields).

Addressed via [f6b7503](https://github.com/Automattic/newspack-plugin/pull/4216/commits/f6b75037fcba00fce613856da8842f2afb641fe9).

miguelpeixe reviewed (approved):
Thank you for revising!

Looks good and tested well
